test/framework/src/main/java/org/elasticsearch/test/rest/yaml/section/ClientYamlTestSection.java

```diff
@@ -98,6 +98,12 @@ public void addExecutableSection(ExecutableSection executableSection) {
                     + "[skip] so runners that do not support the [node_selector] section can skip the test at line ["
                     + doSection.getLocation().lineNumber + "]");
             }
+            if (false == doSection.getApiCallSection().getHeaders().isEmpty()
+                    && false == skipSection.getFeatures().contains("headers")) {
+                throw new IllegalArgumentException("Attempted to add a [do] with a [headers] section without a corresponding "
+                    + "[skip] so runners that do not support the [headers] section can skip the test at line ["
```

Comment:
Maybe worth mentioning the feature has the same name to save people some digging ?
```suggestion
                    + "[skip: {"feature": "headers"}] so runners that do not support the [headers] section can skip the test at line ["
```